File: autojs/src/main/java/autojs/runtime/api/Shell.java
package autojs.runtime.api;

/**
 * Created by Stardust on 2017/3/7.
 */

public class Shell extends com.stardust.scriptdroid.tool.Shell {

    public Shell(boolean root) {
        super(root);
    }

    public void Tap(int x, int y) {
        execute("input tap " + x + " " + y);
    }

    public void Swipe(int x1, int y1, int x2, int y2) {
        execute("input swipe " + x1 + " " + y1 + " " + x2 + " " + y2);
    }

    public void Swipe(int x1, int y1, int x2, int y2, long duration) {
        execute("input swipe " + x1 + " " + y1 + " " + x2 + " " + y2 + " " + duration);
    }

    public void KeyCode(int keyCode) {
        execute("input keyevent " + keyCode);
    }

    public void KeyCode(String keyCode) {
        execute("input keyevent " + keyCode);
    }

    public void Home() {
        KeyCode(3);
    }

    public void Back() {
        KeyCode(4);
    }

    public void Power() {
        KeyCode(26);
    }

    public void Up() {
        KeyCode(19);
    }

    public void Down() {
        KeyCode(20);
    }

    public void Left() {
        KeyCode(21);
    }

    public void Right() {
        KeyCode(22);
    }

    public void OK() {
        KeyCode(23);
    }

    public void VolumeUp() {
        KeyCode(24);
    }

    public void VolumeDown() {
        KeyCode(25);
    }

    public void Menu() {
        KeyCode(1);
    }

    public void Camera() {
        KeyCode(27);
    }

    public void Text(String text) {
        execute("input text " + text);
    }

}

File: autojs/src/main/assets/modules/__shell__.js


module.exports = function(__runtime__, scope){

    scope.SetScreenMetrics = function(w, h){
        __runtime__.getRootShell().SetScreenMetrics(w, h);
    }

    scope.Tap = function(x, y){
        __runtime__.getRootShell().Tap(x, y);
    }

    scope.Swipe = function(x1, y1, x2, y2, duration){
        if(arguments.length == 5){
            __runtime__.getRootShell().Swipe(x1, y1, x2, y2, duration);
        }else{
            __runtime__.getRootShell().Swipe(x1, y1, x2, y2);
        }
    }

    scope.Screencap = function(path){
         __runtime__.getRootShell().Screencap(path);
    }

    scope.KeyCode = function(keyCode){
        __runtime__.getRootShell().KeyCode(keyCode);
    }

    scope.Home = function(){
        return KeyCode(3);
    }

    scope.Back = function(){
        return KeyCode(4);
    }

    scope.Power = function(){
        return KeyCode(26);
    }

    scope.Up = function(){
        return KeyCode(19);
    }

    scope.Down = function(){
        return KeyCode(20);
    }

    scope.Left = function(){
        return KeyCode(21);
    }

    scope.Right = function(){
        return KeyCode(22);
    }

    scope.OK = function(){
        return KeyCode(23);
    }

    scope.VolumeUp = function(){
        return KeyCode(24);
    }

    scope.VolumeDown = function(){
        return KeyCode(25);
    }

    scope.Menu = function(){
        return KeyCode(1);
    }

    scope.Camera = function(){
        return KeyCode(27);
    }

    scope.Text = function(text){
         __runtime__.getRootShell().Text(text);
    }

    scope.Input = scope.Text;

    return function(cmd, root){
       root = root ? 1 : 0;
       return __runtime__.shell(cmd, root);
   };
}
